Algorithms/Sorting/bucket_sort.cpp

```diff
@@ -0,0 +1,88 @@
+// Name : Anjali Kumari
+// Email : LIT2016033@iiitl.ac.in
+
+// This program contains implementation of Bucket Sort Algorithm.
+
+#include<iostream>
+#include<algorithm>
+#include<vector>
+#include<utility>
+using namespace std;
+
+#define bucket_size 10
+
+void bucketSort(float arr[], long n)
+{  
+    vector<float> bucket[bucket_size];
+    //we make empty bucket vectors of size bucket_size , vector is choosen instead of
+	//2-d array because we can dynamically regulate size of vector.
+	
+    for(long i=0;i<n;i++)
+    {
+       long d = arr[i]*n; //Index determination in a bucket.
+       bucket[d].push_back(arr[i]);
+    }
+    for(long i=0;i<bucket_size;i++)
+    {
+       	sort(bucket[i].begin(),bucket[i].end());
+    }
+    long j=0,index=0;
+    while(j!=bucket_size)
+    {
+       for(long i=0;i<bucket[j].size();i++)
+       {
+            arr[index++]=bucket[j][i];
+       }
+       j++;
+    }
+}
+ 
+//Function to print the elements of Array
+void printArray(float arr[],long arr_length)
+{
+    for(long i=0; i<arr_length; i++)
+    {
+      printf("%f ", arr[i]);
+    }
+    printf("\n");
+}
+
+// Driver program to test bucketSort function
+int main()
+{
+    long n;
+    printf("Please enter the number of elements in Array."); 
+    scanf("%lld",&n);
```

Comment:
Since `n` is of type long, it should be `scanf("%ld", &n)`. Its giving an warning while running the warning, so better if you change it.